librz/core/cmd_analysis.c

```diff
@@ -7775,144 +7626,193 @@ static void cmd_agraph_node(RzCore *core, const char *input) {
 	}
 }
 
+RZ_IPI void rz_core_agraph_add_edge(RzCore *core, const char *un, const char *vn) {
+	RzANode *u = rz_agraph_get_node(core->graph, un);
+	RzANode *v = rz_agraph_get_node(core->graph, vn);
+	if (!u) {
+		rz_cons_printf("Node %s not found!\n", un);
+		return;
+	} else if (!v) {
+		rz_cons_printf("Node %s not found!\n", vn);
+		return;
+	}
+	rz_agraph_add_edge(core->graph, u, v);
+}
+
+RZ_IPI void rz_core_agraph_del_edge(RzCore *core, const char *un, const char *vn) {
+	RzANode *u = rz_agraph_get_node(core->graph, un);
+	RzANode *v = rz_agraph_get_node(core->graph, vn);
+	if (!u) {
+		rz_cons_printf("Node %s not found!\n", un);
+		return;
+	} else if (!v) {
+		rz_cons_printf("Node %s not found!\n", vn);
+		return;
+	}
+	rz_agraph_del_edge(core->graph, u, v);
+}
+
 static void cmd_agraph_edge(RzCore *core, const char *input) {
+	char **args;
+	int n_args;
+
 	switch (*input) {
 	case ' ': // "age"
-	case '-': { // "age-"
-		RzANode *u, *v;
-		char **args;
-		int n_args;
-
 		args = rz_str_argv(input + 1, &n_args);
 		if (n_args != 2) {
 			rz_cons_printf("Wrong arguments\n");
 			rz_str_argv_free(args);
 			break;
 		}
 
-		u = rz_agraph_get_node(core->graph, args[0]);
-		v = rz_agraph_get_node(core->graph, args[1]);
-		if (!u || !v) {
-			if (!u) {
-				rz_cons_printf("Node %s not found!\n", args[0]);
-			} else {
-				rz_cons_printf("Node %s not found!\n", args[1]);
-			}
+		rz_core_agraph_add_edge(core, args[0], args[1]);
+		rz_str_argv_free(args);
+		break;
+	case '-': // "age-"
+		args = rz_str_argv(input + 1, &n_args);
+		if (n_args != 2) {
+			rz_cons_printf("Wrong arguments\n");
 			rz_str_argv_free(args);
 			break;
 		}
-		if (*input == ' ') {
-			rz_agraph_add_edge(core->graph, u, v);
-		} else {
-			rz_agraph_del_edge(core->graph, u, v);
-		}
+
+		rz_core_agraph_del_edge(core, args[0], args[1]);
 		rz_str_argv_free(args);
 		break;
-	}
 	case '?': // "age?"
 	default:
 		rz_core_cmd_help(core, help_msg_age);
 		break;
 	}
 }
 
+RZ_IPI void rz_core_agraph_print_custom(RzCore *core) {
```

Comment:
I suggest putting those into the separate file, e.g. `librz/core/cagraph.c` or something like that.